server_test.go

```diff
@@ -4,33 +4,44 @@ package main
 
 import "testing"
 
+var colortests = []struct {
+	in    string
+	valid bool // If valid format
+	R     byte
+	G     byte
+	B     byte
+}{
+	{"#123", false, 0, 0, 0},
+	{"#1234567", false, 0, 0, 0},
+	{"$123456", false, 0, 0, 0},
+	{"#12345+", false, 0, 0, 0},
+	{"#fFGG00", false, 0, 0, 0},
+	{"", false, 0, 0, 0},
+	{"#123456", true, 0x12, 0x34, 0x56},
+	{"#C0FfeE", true, 0xc0, 0xff, 0xee},
+}
+
 func TestParseHexColor(t *testing.T) {
-	empty := ""
-	color, err := parseHexColor(empty)
-	if err == nil {
-		t.Fatalf("Empty color string should return error, but did not")
-	}
+	// Perform the table driven tests
+	for _, ct := range colortests {
 
-	tooLong := "#1234567"
-	color, err = parseHexColor(tooLong)
-	if err == nil {
-		t.Fatalf("Invalid color string %s should return error, but did not",
-			tooLong)
-	}
+		color, err := parseHexColor(ct.in)
+		if !ct.valid && err == nil {
+			if ct.in != "" {
+				t.Fatalf("Invalid color string %s should return "+
```

Comment:
Just do `t.Fatalf("Invalid color string <%s> should return "...` and you don't have to special case the empty string case.

Reply:
@halseth Thanks for the review. I've included your suggestions.